Model/Sync/Catalog/Processor.php

```diff
@@ -173,6 +173,17 @@ public function process($forceSyncProducts = [], $storeIds = [])
                                 $this->coreConfig->getStoreName($storeId) . PHP_EOL;
                         }
                     }
+                    if ($this->coreConfig->isSyncResetInProgress($storeId, 'catalog')) {
+                        $disabled = true;
+                        $this->yotpoCatalogLogger->info(
+                            __(
+                                'Product sync is skipped because catalog sync reset is in progress
+                        - Magento Store ID: %1, Name: %2',
+                                $storeId,
+                                $this->coreConfig->getStoreName($storeId)
+                            )
+                        );
+                    }
```

Comment:
This can be checked at the beginning of the loop, why positioning it over here?

Reply:
Okay, we will make these changes